src/IO/S3/PocoHTTPClient.cpp

```diff
@@ -138,6 +202,8 @@ void PocoHTTPClient::MakeRequestInternal(
                 uri = location;
                 LOG_DEBUG(log, "Redirecting request to new location: {}", location);
```

Comment:
There must be an option to disable redirects, otherwise malicious S3 server (by user-provided URL) can perform SSRF attacks.

Reply:
But it will go to another pull request...

Reply:
I've made a new issue for this (https://github.com/ClickHouse/ClickHouse/issues/12531)